File: src/main/java/korebit/model/beans/sub/ProductSupplierId.java
package korebit.model.beans.sub;

import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serial;
import java.io.Serializable;

@Embeddable
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class ProductSupplierId implements Serializable {
    @Serial
    private static final long serialVersionUID = 1L;

    private Long productId;
    private Long supplierId;
}

File: src/main/kotlin/atlix/model/content/ProductSupplierId.kt
package atlix.model.content

import jakarta.persistence.Column
import jakarta.persistence.Embeddable

@Embeddable
data class ProductSupplierId(
    @Column(name = "product_id")
    var productId: Long = 0,

    @Column(name = "supplier_id")
    var supplierId: Int = 0
) : java.io.Serializable
